### PR TITLE
GA chatMessageHostedContent for channel chatMessage

### PR DESCRIPTION
**Pull request by satadrubiswasmsft**
GA chatMessageHostedContent for channel chatMessage

### CODE REVIEW

File: api-reference/v1.0/api/chatmessage-list-chatmessagehostedcontents.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+---
+title: "List chatMessageHostedContents"
```
Comment (Lauragra) [inline api-reference/v1.0/api/chatmessage-list-chatmessagehostedcontents.md:2]: The file name here should be chatmessage-list-hostedcontents, and the title field should be List hostedContents, to match the H1. Please fix this in the beta topic as well.

File: api-reference/v1.0/api/chatmessage-list-chatmessagehostedcontents.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+If successful, this method returns a `200 OK` response code and a collection of [chatMessageHostedContent](../resources/chatmessagehostedcontent.md) objects in the response body.
+
+## Examples
+
+### Request
+
+The following is an example of the request.
+
+# [HTTP](#tab/http)
```
Comment (Lauragra) [inline api-reference/v1.0/api/chatmessage-list-chatmessagehostedcontents.md:63]: Global: Please delete all the H1s here and the include links to the snippets (line 63 and lines 73-88). Those will get added via the weekly automation when this PR is merged into master.

File: api-reference/v1.0/includes/snippets/csharp/get-chatmessagehostedcontent-csharp-snippets.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+description: "Automatically generated file. DO NOT MODIFY"
```
Comment (Lauragra) [inline api-reference/v1.0/includes/snippets/csharp/get-chatmessagehostedcontent-csharp-snippets.md:2]: Global: Please remove the snippet files; these will be added via automation.

**Review by Lauragra — CHANGES_REQUESTED** [review]
Please see comments and include an update to the changelog JSON file with this PR.

File: api-reference/v1.0/api/chatmessage-list-chatmessagehostedcontents.md (hunk 1 above)
Comment (satadrubiswasmsft) [reply api-reference/v1.0/api/chatmessage-list-chatmessagehostedcontents.md:2]: > The file name here should be chatmessage-list-hostedcontents, and the title field should be List hostedContents, to match the H1. Please fix this in the beta topic as well.

@Lauragra  should the name be `chatmessage-list-hostedcontents` (suggested by you) OR `chatmessage-list-chatmessagehostedcontents` (the actual filename)?

Comment (satadrubiswasmsft) [reply api-reference/v1.0/api/chatmessage-list-chatmessagehostedcontents.md:2]: Ignore my comment. I see what you mean

Comment (satadrubiswasmsft) [reply api-reference/v1.0/api/chatmessage-list-chatmessagehostedcontents.md]: fixed

File: api-reference/v1.0/includes/snippets/csharp/get-chatmessagehostedcontent-csharp-snippets.md (hunk 3 above)
Comment (satadrubiswasmsft) [reply api-reference/v1.0/includes/snippets/csharp/get-chatmessagehostedcontent-csharp-snippets.md]: fixed

**Comment by satadrubiswasmsft** [comment]
> Please see comments and include an update to the changelog JSON file with this PR.

@Lauragra Should the `changelog/Microsoft.Teams.Core.json` updated by hand? Or is there a script that will append the changes to it?

**Review by Lauragra — CHANGES_REQUESTED** [review]
Please surface these APIs in the TOC. It looks like they are not exposed in the beta TOC and need to be added to v1.0 as well.

**Comment by satadrubiswasmsft** [comment]
> Please surface these APIs in the TOC. It looks like they are not exposed in the beta TOC and need to be added to v1.0 as well.

Updated the TOC. The beta was updated but the file name was different. Fixed that

**Review by Lauragra — APPROVED** [review]

File: api-reference/beta/resources/chatmessagehostedcontent.md [hunk 4]
```diff
@@ -21,7 +21,7 @@ they are stored in SharePoint or OneDrive.
 
 | Method       | Return Type | Description |
 |:-------------|:------------|:------------|
-| [List chatMessageHostedContent](../api/chatmessage-list-chatmessagehostedcontents.md) | [chatMessageHostedContent](chatmessagehostedcontent.md) | Retrieve the list of **chatMessageHostedContent** for a message. |
+| [List chatMessageHostedContent](../api/chatmessage-list-hostedcontents.md) | [chatMessageHostedContent](chatmessagehostedcontent.md) | Retrieve the list of **chatMessageHostedContent** for a message. |
```
Comment (AkJo) [inline api-reference/beta/resources/chatmessagehostedcontent.md:24]: [chatMessageHostedContent](chatmessagehostedcontent.md) collection

File: api-reference/v1.0/api/chatmessage-list-hostedcontents.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+> [!NOTE]
+> Before calling this API with application permissions, you must request access. For details, see [Protected APIs in Microsoft Teams](/graph/teams-protected-apis).
+
+## HTTP request
+
+<!-- { "blockType": "ignored" } -->
+
+```http
+GET /teams/{id}/channels/{id}/messages/{id}/hostedContents
```
Comment (AkJo) [inline api-reference/v1.0/api/chatmessage-list-hostedcontents.md:36]: /teams/{team-id}/channels/{channel-id}/messages/{message-id}/hostedContents

File: api-reference/v1.0/api/chatmessage-list-hostedcontents.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+The following is an example of the request.
+
+<!-- {
+  "blockType": "request",
+  "name": "get_hostedcontents"
+}-->
+
+```msgraph-interactive
+GET https://graph.microsoft.com/v1.0/teams/{id}/channels/{id}/messages/{id}/hostedContents
```
Comment (AkJo) [inline api-reference/v1.0/api/chatmessage-list-hostedcontents.md:69]: Please use illustrative values in examples.

Replace place holders with real values.

File: api-reference/v1.0/api/chatmessage-list-hostedcontents.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+  "@odata.type": "microsoft.graph.chatMessageHostedContent",
+  "isCollection": true
+} -->
+
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+
+{
+  "value": [
```
Comment (AkJo) [inline api-reference/v1.0/api/chatmessage-list-hostedcontents.md:92]: Same comment. Please provide a real response payload.

File: api-reference/v1.0/api/chatmessagehostedcontent-get.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+> [!NOTE]
+> Before calling this API with application permissions, you must request access. For details, see [Protected APIs in Microsoft Teams](/graph/teams-protected-apis).
+
+## HTTP request
+
+<!-- { "blockType": "ignored" } -->
+
+```http
+GET /teams/{id}/channels/{id}/messages/{id}/hostedContents/{id}
```
Comment (AkJo) [inline api-reference/v1.0/api/chatmessagehostedcontent-get.md:36]: Same comment.

Comment (AkJo) [reply api-reference/v1.0/api/chatmessagehostedcontent-get.md:36]: /teams/{team-id}/channels/{channel-id}/messages/{message-id}/hostedContents/{hosted-content-id}

File: api-reference/v1.0/api/chatmessagehostedcontent-get.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+The following is an example of the request.
+
+<!-- {
+  "blockType": "request",
+  "name": "get_chatmessagehostedcontent"
+}-->
+
+```msgraph-interactive
+GET https://graph.microsoft.com/v1.0/teams/{id}/channels/{id}/messages/{id}/hostedContents/{id}
```
Comment (AkJo) [inline api-reference/v1.0/api/chatmessagehostedcontent-get.md:71]: Illustrative values.

File: api-reference/v1.0/api/chatmessagehostedcontent-get.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+#### Request
+
+
+<!-- {
+  "blockType": "request",
+  "name": "get_chatmessagehostedcontent"
+}-->
+```msgraph-interactive
+GET https://graph.microsoft.com/v1.0/teams/{id}/channels/{id}/messages/{id}/hostedContents/{id}
```
Comment (AkJo) [inline api-reference/v1.0/api/chatmessagehostedcontent-get.md:106]: Illustrative values.

File: api-reference/v1.0/api/chatmessagehostedcontent-get.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+  "truncated": true,
+  "@odata.type": "microsoft.graph.chatMessageHostedContent"
+} -->
+
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+
+{
+  "id": "id-value"
```
Comment (AkJo) [inline api-reference/v1.0/api/chatmessagehostedcontent-get.md:92]: Illustrative values.

File: api-reference/v1.0/resources/chatmessagehostedcontent.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+Represents Teams content hosted in a chat message, such as images or code snippets.
+[File attachments](chatmessageattachment.md) are not hosted content;
+they are stored in SharePoint or OneDrive.
+
+## Methods
+
+| Method       | Return Type | Description |
+|:-------------|:------------|:------------|
+| [List chatMessageHostedContent](../api/chatmessage-list-hostedcontents.md) | [chatMessageHostedContent](chatmessagehostedcontent.md) | Retrieve the list of **chatMessageHostedContent** for a message. |
```
Comment (AkJo) [inline api-reference/v1.0/resources/chatmessagehostedcontent.md:22]: collection

Comment (AkJo) [inline api-reference/v1.0/resources/chatmessagehostedcontent.md:22]: List hosted content

Comment (AkJo) [reply api-reference/v1.0/resources/chatmessagehostedcontent.md:22]: Please make this change in the beta doc as well.

File: api-reference/v1.0/resources/chatmessagehostedcontent.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+| [Get chatMessageHostedContent](../api/chatmessagehostedcontent-get.md) | [chatMessageHostedContent](chatmessagehostedcontent.md) | Read the properties and relationships of a **chatMessageHostedContent** object. |
```
Comment (AkJo) [inline api-reference/v1.0/resources/chatmessagehostedcontent.md:23]: Get content hosted in chat message

Please make this change in beta documentation too.

File: api-reference/v1.0/resources/chatmessagehostedcontent.md [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+|contentType   |String       | When posting new chat message hosted content, represents the type of content, such as image/png. Write-only. |
+
+### Instance attributes
+
+Instance attributes are properties with special behaviors.
+These properties are temporary and either define behavior the service should perform or provide short-term property values, like a download URL for an item that expires.
+
+| Property name                     | Type   | Description
+|:----------------------------------|:-------|:--------------------------------
+| @microsoft.graph.temporaryId      | string | Represents the temporaryId for the hosted content while posting a message to refer to the hosted content in **chatMessage** resource being sent. Write-only. |
```
Comment (AkJo) [inline api-reference/v1.0/resources/chatmessagehostedcontent.md:40]: Please add deeplink to chatMessage page.

File: api-reference/v1.0/api/chatmessage-list-hostedcontents.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
+---
+title: "List hostedContents"
```
Comment (AkJo) [inline api-reference/v1.0/api/chatmessage-list-hostedcontents.md:2]: List content hosted in chat message

File: api-reference/v1.0/api/chatmessage-list-hostedcontents.md [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+description: "Retrieve the list of chatMessageHostedContent objects from a message."
+localization_priority: Normal
+author: "clearab"
+ms.prod: "microsoft-teams"
+doc_type: "apiPageType"
+---
+
+# List hostedContents
```
Comment (AkJo) [inline api-reference/v1.0/api/chatmessage-list-hostedcontents.md:10]: List content hosted in chat message

File: api-reference/beta/api/chatmessage-list-hostedcontents.md [hunk 17]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "List chatMessageHostedContents"
+title: "List hostedContents"
```
Comment (AkJo) [inline api-reference/beta/api/chatmessage-list-hostedcontents.md:2]: List content hosted inside chat message

File: api-reference/beta/api/chatmessage-list-hostedcontents.md [hunk 18]
```diff
@@ -1,5 +1,5 @@
...
 description: "Retrieve the list of chatMessageHostedContent objects from a message."
 localization_priority: Normal
 author: "clearab"
```
Comment (AkJo) [inline api-reference/beta/api/chatmessage-list-hostedcontents.md:5]: Please change this to "AkJo"

File: api-reference/beta/resources/chatmessagehostedcontent.md (hunk 4 above)
Comment (satadrubiswasmsft) [reply api-reference/beta/resources/chatmessagehostedcontent.md:24]: This is contradicting what @Lauragra had suggested

Comment (satadrubiswasmsft) [reply api-reference/beta/resources/chatmessagehostedcontent.md:24]: also the file name was changed to `chatmessage-list-hostedcontents.md`

**Comment by jasonjoh** [comment]
Please disregard the earlier message from msftbot. Your pull request was incorrectly flagged.

**Comment by lramosvea** [comment]
@satadrubiswasmsft have all comments from the reviewers been addressed? If so, please add the "content review" label to this PR.

File: api-reference/beta/api/chatmessage-list-hostedcontents.md [hunk 19]
```diff
@@ -1,8 +1,8 @@
 ---
-title: "List chatMessageHostedContents"
+title: "List content hosted inside chat message"
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessage-list-hostedcontents.md:2]: This title should be List hostedContents; it should match the H1.

File: api-reference/beta/resources/chatmessagehostedcontent.md [hunk 20]
```diff
@@ -21,8 +21,8 @@ they are stored in SharePoint or OneDrive.
 
 | Method       | Return Type | Description |
 |:-------------|:------------|:------------|
-| [List chatMessageHostedContent](../api/chatmessage-list-chatmessagehostedcontents.md) | [chatMessageHostedContent](chatmessagehostedcontent.md) | Retrieve the list of **chatMessageHostedContent** for a message. |
-| [Get chatMessageHostedContent](../api/chatmessagehostedcontent-get.md) | [chatMessageHostedContent](chatmessagehostedcontent.md) | Read the properties and relationships of a **chatMessageHostedContent** object. |
+| [List content hosted in chat message](../api/chatmessage-list-hostedcontents.md) | [chatMessageHostedContent](chatmessagehostedcontent.md) | Retrieve the collection of **chatMessageHostedContent** for a message. |
```
Comment (Lauragra) [inline api-reference/beta/resources/chatmessagehostedcontent.md:24]: This should be List hostedContents.

File: api-reference/beta/resources/chatmessagehostedcontent.md [hunk 21]
```diff
@@ -21,8 +21,8 @@ they are stored in SharePoint or OneDrive.
...
+| [Get content hosted in chat message](../api/chatmessagehostedcontent-get.md) | [chatMessageHostedContent](chatmessagehostedcontent.md) | Read the properties and relationships of a **chatMessageHostedContent** object. |
```
Comment (Lauragra) [inline api-reference/beta/resources/chatmessagehostedcontent.md:25]: This should be Get chatMessageHostedContent.

File: api-reference/v1.0/api/chatmessage-list-hostedcontents.md [hunk 22]
```diff
@@ -0,0 +1,112 @@
+---
+title: "List content hosted inside chat message"
```
Comment (Lauragra) [inline api-reference/v1.0/api/chatmessage-list-hostedcontents.md:2]: Change the title and H1 to List hostedContents.

File: api-reference/v1.0/api/chatmessage-list-hostedcontents.md [hunk 23]
```diff
@@ -0,0 +1,112 @@
...
+GET https://graph.microsoft.com/v1.0/teams/3ade275a-77cb-444e-904d-9cb268c24b9d/channels/19%3A7d8190d723f745c38750a6957777u8a8%40thread.skype/messages/1602725975403/hostedContents
+```
+
+
+### Response
+
+The following is an example of the response.
+
+> [!NOTE]
+> The response object shown here might be shortened for readability. All the properties will be returned from an actual call.
```
Comment (Lauragra) [inline api-reference/v1.0/api/chatmessage-list-hostedcontents.md:78]: Global: Please change the note formatting to >**Note:** and remove the second sentence from the boilerplate text.

File: api-reference/v1.0/api/chatmessage-list-hostedcontents.md [hunk 24]
```diff
@@ -0,0 +1,112 @@
...
+
+## Optional query parameters
+
+This operation does not support the [OData query parameters](/graph/query-parameters) to customize the response.
+
+## Request headers
+
+| Name      |Description|
+|:----------|:----------|
+| Authorization | Bearer {code} |
```
Comment (Lauragra) [inline api-reference/v1.0/api/chatmessage-list-hostedcontents.md:47]: Please add Required to the description.

File: api-reference/v1.0/resources/chatmessagehostedcontent.md [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+
+Represents Teams content hosted in a chat message, such as images or code snippets.
+[File attachments](chatmessageattachment.md) are not hosted content;
+they are stored in SharePoint or OneDrive.
+
+## Methods
+
+| Method       | Return Type | Description |
+|:-------------|:------------|:------------|
+| [List content hosted in chat message](../api/chatmessage-list-hostedcontents.md) | [chatMessageHostedContent](chatmessagehostedcontent.md) | Retrieve the collection of **chatMessageHostedContent** for a message. |
```
Comment (Lauragra) [inline api-reference/v1.0/resources/chatmessagehostedcontent.md:22]: Please update the method names here per previous comments.

**Pull request closed without merging**